### ISSUE TITLE
Support authenticating as a Google+ Page

### ISSUE THREAD

**Issue by xd1936** [google/google-api-javascript-client#236]
I'm trying to use the Google JavaScript Client API Library to authenticate and interact with the YouTube API on an internal web tool. The person signing in is a Google user who is a manager of several Google+ pages, and those [pages are owners of YouTube channels](https://support.google.com/youtube/answer/4642409).

I want the user to click my sign-in button on my page, authenticate as their Google+ page, and then my code preforms API calls to that page's YouTube channel.

Right now, I have something like this:

``` javascript
    var clientId = 'my-client-id';
    var apiKey = 'my-api-key';
    var scopes = 'profile https://www.googleapis.com/auth/plus.login https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/youtube';

    var signinButton = document.getElementById('signin');
    var signoutButton = document.getElementById('signout');

    function initAuth() {
        gapi.client.setApiKey(apiKey);
        gapi.auth2.init({
            client_id: clientId,
            scope: scopes
        }).then(function() {
            // Listen for sign-in state changes.
            gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);

            // Handle the initial sign-in state.
            updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());

            signinButton.addEventListener("click", handleSigninClick);
            signoutButton.addEventListener("click", handleSignoutClick);
        });
    }

    function updateSigninStatus(isSignedIn) {
        if (isSignedIn) {
            signinButton.style.display = 'none';
            signoutButton.style.display = 'block';
        } else {
            signinButton.style.display = 'block';
            signoutButton.style.display = 'none';
        }
    }

    function handleSigninClick(event) {
        gapi.auth2.getAuthInstance().signIn();
    }

    function handleSignoutClick(event) {
        gapi.auth2.getAuthInstance().signOut();
    }

    // Load the API client and auth library
    gapi.load('client:auth2', initAuth);
```

The buttons work perfectly. When my user clicks on the `<button id="signin"></button>`, it brings up the Google Auth dialog. If they authenticate as themselves, everything works perfectly. However, **if they authenticate as a Google+ Page that they own, I get a console error:**

![Uncaught Object object on cb=gapi.loaded_0:51](http://i.stack.imgur.com/gyB3a.png)
## TL;DR:

Does anyone know if the Google Javascript Client API library supports authenticating as a Google+ Page, and if so, how I should be handling what is being returned?

Thanks!

**Comment by cyberaleks** [comment]
+1. Same issue, when i use gapi.auth2. The error is "failedToken" and message is "USER_LOGGED_OUT".  When i use deprecated gapi.auth, it works.

**Comment by PatTheMav** [comment]
+1 - I've been tearing my hairs out over this one - no official word on this anywhere, just 3 posts on Stackoverflow with everyone asking whether this was fixed - it's really bad.

Especially as it seems to work in Google's own API explorer or oAUTH2 test playground, but not if used using the JS API.

**Comment by TMSCH** [comment]
@xd1936 @cyberaleks @PatTheMav we are working on addressing this issue. I'll update this thread when I have more information.

**Comment by xd1936** [comment]
@TMSCH So glad to hear that! Looking forward to your updates.

**Comment by TMSCH** [comment]
@xd1936 @cyberaleks @PatTheMav please try again, we have released a fix a couple of weeks ago that should address the issue. Thanks!

**Comment by PatTheMav** [comment]
I've moved the whole authentication stack into the backend, but revived some old frontend-code and authentication and pulling data as a Google+ page seemed to work as intended.. 👍

**Comment by TMSCH** [comment]
Thanks @PatTheMav for testing! I'm glad it's working now, but sorry we couldn't make it work in time for you to keep your original implementation.

**Comment by xd1936** [comment]
Working perfectly for me! Thanks so much, @TMSCH!

**Comment by TMSCH** [comment]
Great, thanks @xd1936! Closing this issue now as fixed.
